artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/server/impl/RemotingServiceImpl.java

```diff
@@ -408,27 +407,15 @@ public void stop(final boolean criticalError) throws Exception {
          conn.disconnect(criticalError);
       }
 
-      Map<Acceptor, Future<?>> acceptorFutures = new HashMap<>();
       for (Acceptor acceptor : acceptors.values()) {
          try {
-            acceptorFutures.put(acceptor, acceptor.asyncStop());
+            acceptor.stop();
          } catch (Throwable t) {
             ActiveMQServerLogger.LOGGER.errorStoppingAcceptor(acceptor.getName());
          }
       }
 
-      for (Map.Entry<Acceptor, Future<?>> acceptorFuture : acceptorFutures.entrySet()) {
-         if (acceptorFuture.getValue() != null) {
-            try {
-               acceptorFuture.getValue().get();
```

Comment:
@brusdev there's probably some issue with some Failover tests on getting the return here without a timeout.

Probably a timeout of 1 sec here would suffice, as we just want to make sure we actually close the acceptors.

Reply:
I totally agree on the need for a timeout. On my machine it takes 1.5 seconds in the worst case, wdyt about a total timeout of 3 seconds?

An example just to clarify what I mean:

>       long acceptorTimeout = System.currentTimeMillis() + 3000;
>       for (Map.Entry<Acceptor, Future<?>> acceptorFuture : acceptorFutures.entrySet()) {
>          if (acceptorFuture.getValue() != null) {
>             try {
>                acceptorFuture.getValue().get(Math.max(0, acceptorTimeout - System.currentTimeMillis()), TimeUnit.MILLISECONDS);
>             } catch (Throwable t) {
>                ActiveMQServerLogger.LOGGER.errorStoppingAcceptor(acceptorFuture.getKey().getName());
>             }
>          }
>       }
>